### PR TITLE
[WIP] GPU/TPU resource-collector add-on

### PR DESCRIPTION
**Pull request by z1ens**
- Add GPU/TPU addonplacementscore to each node in the managed clusters.
- Develop the add-on in addontemplate version
- Create a comprehensive user guide
REF: https://github.com/open-cluster-management-io/ocm/issues/369

### CODE REVIEW

File: resource-usage-collect-addon/deploy/resources/addon-template.yaml [hunk 1]
```diff
@@ -0,0 +1,112 @@
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: AddOnTemplate
+metadata:
+  name: resource-usage-collect
+spec:
+  addonName: resource-usage-collect
+  agentSpec:
+    workload:
+      manifests:
+        - kind: ClusterRole
```
Comment (haoqing0110) [inline resource-usage-collect-addon/deploy/resources/addon-template.yaml:10]: Could you try to minimize the rbac permission?

File: resource-usage-collect-addon/deploy/resources/managed-cluster-set-binding.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+kind: ManagedClusterSetBinding
+metadata:
+  generation: 1
+  name: global
+  namespace: default
+spec:
+  clusterSet: global
+
+---
+apiVersion: cluster.open-cluster-management.io/v1beta1
```
Comment (haoqing0110) [inline resource-usage-collect-addon/deploy/resources/managed-cluster-set-binding.yaml:11]: suggest to move placement to a seperate yaml.

**Comment by haoqing0110** [comment]
- remove the binary `resource-usage-collect-addon/addon` 
- refer to https://github.com/open-cluster-management-io/addon-contrib/pull/14 , can try to use sdk to generate scores.

File: resource-usage-collect-addon/Makefile [hunk 3]
```diff
@@ -19,9 +19,9 @@ PWD=$(shell pwd)
 
 # Image URL to use all building/pushing image targets;
 GO_BUILD_PACKAGES :=./pkg/...
-IMAGE ?= resource-usage-collect-addon
-IMAGE_REGISTRY ?= quay.io/haoqing
-IMAGE_TAG ?= latest
+IMAGE ?= resource-usage-collect-addon-template
```
Comment (zhujian7) [inline resource-usage-collect-addon/Makefile:22]: It is not necessary to add the `-template` suffix.

File: resource-usage-collect-addon/deploy/resources/managed-cluster-set-binding.yaml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+apiVersion: cluster.open-cluster-management.io/v1beta2
+kind: ManagedClusterSetBinding
+metadata:
+  generation: 1
```
Comment (zhujian7) [inline resource-usage-collect-addon/deploy/resources/managed-cluster-set-binding.yaml:4]: the generation  line is not needed.

File: resource-usage-collect-addon/deploy/resources/cluster-management-addon.yaml [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  supportedConfigs:
+    - group: addon.open-cluster-management.io
+      resource: addontemplates
+      defaultConfig:
+        name: resource-usage-collect
+  installStrategy:
+    type: Placements
+    placements:
+      - name: placement-all
+        namespace: default
```
Comment (zhujian7) [inline resource-usage-collect-addon/deploy/resources/cluster-management-addon.yaml:17]: EOF, can we add a blank line for all yaml files?

**Comment by zhujian7** [comment]
it seems the DCO check is not working now. I will try to fix it.

**Comment by zhujian7** [comment]
> it seems the DCO check is not working now. I will try to fix it.

A DCO check has been added, @z1ens could you rebase and sign off your commits?

File: resource-usage-collect-addon/deploy/resources/managed-cluster-set-binding.yaml (hunk 2 above)
Comment (z1ens) [reply resource-usage-collect-addon/deploy/resources/managed-cluster-set-binding.yaml:11]: got it

File: .gitignore [hunk 6]
```diff
@@ -12,4 +12,5 @@
 *.out
 
 # Dependency directories (remove the comment below to include it)
-# vendor/
+# vendor/.DS_Store
```
Comment (zhujian7) [inline .gitignore:15]: should `.DS_Store` be in a new line? It is not a sub folder of the vendor, right?

**Pull request closed without merging**